src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,247 @@
+    }
+
+    /**
+     * Creates a new ToDo object.
+     *
+     * @return a ToDo object.
+     * @throws MissingTaskDescriptionException Exception is thrown when task description is missing.
+     */
+    public Task createNewToDo() throws MissingTaskDescriptionException {
+        try {
+
+            Task newToDoTask = new ToDo(commandParts[1]);
+            return newToDoTask;
+
+        } catch (Exception e) {
+            throw new MissingTaskDescriptionException(HORIZONTAL_LINE
+                    + "\r\n"
+                    + "Oops! The description cannot be empty :("
+                    + "\r\n"
+                    + HORIZONTAL_LINE);
+        }
+    }
+
+    /**
+     * Creates a new Deadline object.
+     *
+     * @return a Deadline object.
+     * @throws MissingTaskDescriptionException Exception is thrown when task description is missing.
+     */
+    public Task createNewDeadline() throws MissingTaskDescriptionException {
+        try {
+
+            String[] deadlineParts = commandParts[1].split("/by");
+            LocalDate deadline = LocalDate.parse(deadlineParts[1].trim());
+            String afterDateTimeFormat = deadline.format(DateTimeFormatter.ofPattern("dd MMM yyyy"));
+            Task newDeadlineTask = new Deadline(deadlineParts[0].trim(), afterDateTimeFormat);
+            return newDeadlineTask;
+
+        } catch (Exception e) {
+
+            throw new MissingTaskDescriptionException(HORIZONTAL_LINE
+                    + "\r\n"
+                    + "Oops! The description cannot be empty :("
+                    + "\r\n"
+                    + HORIZONTAL_LINE);
+        }
+    }
+
+    /**
+     * Creates a new Event object.
+     *
+     * @return a Event object.
+     * @throws MissingTaskDescriptionException Exception is thrown when task description is missing.
+     */
+    public Task createNewEvent() throws MissingTaskDescriptionException {
+        try {
+
+            String[] eventParts = commandParts[1].split("/at");
+            LocalDate event = LocalDate.parse(eventParts[1].trim());
+            String afterDateTimeFormat = event.format(DateTimeFormatter.ofPattern("dd MMM yyyy"));
+            Task newEventTask = new Event(eventParts[0].trim(), afterDateTimeFormat);
+            return newEventTask;
+
+        } catch (Exception e) {
+
+            throw new MissingTaskDescriptionException(HORIZONTAL_LINE
+                    + "\r\n"
+                    + "Oops! The description cannot be empty :("
+                    + "\r\n"
+                    + HORIZONTAL_LINE);
+        }
+    }
+
+    /**
+     * Returns the task number of the task for its status to be updated.
+     *
+     * @return the task number of the task.
+     * @throws MissingTaskNumberException  Exception is thrown when task number is missing.
+     */
+    public int getDoneTaskNumber() throws MissingTaskNumberException {
+
+        int taskNumber = -1;
+        taskNumber = markDoneTask();
+        return taskNumber;
```

Comment:
i realised that for some of your methods you initialised your taskNumber as -1 first. im not quite sure why you did that and there might be a reason so **please ignore this comment if there is a reason**. but i feel like the initialisation might be redundant? you could probably just shrink this method down to:

public int ........... {
    return markDoneTask();
}

just a suggestion!